controller/backup_controller.go

```diff
@@ -444,6 +439,106 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 	return nil
 }
 
+// handleAttachmentTicketDeletion check and delete attachment so that the source volume is detached if needed
+func (bc *BackupController) handleAttachmentTicketDeletion(backup *longhorn.Backup, volumeName string) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "handleAttachmentTicketDeletion: failed to clean up attachment")
+	}()
+
+	va, err := bc.ds.GetLHVolumeAttachment(types.GetLHVolumeAttachmentNameFromVolumeName(volumeName))
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	attachmentTicketID := longhorn.GetAttachmentTicketID(longhorn.AttacherTypeBackupController, backup.Name)
+
+	if _, ok := va.Spec.AttachmentTickets[attachmentTicketID]; ok {
+		delete(va.Spec.AttachmentTickets, attachmentTicketID)
+		if _, err = bc.ds.UpdateLHVolumeAttachment(va); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// handleAttachmentTicketCreation check and create attachment so that the source volume is attached if needed
+func (bc *BackupController) handleAttachmentTicketCreation(backup *longhorn.Backup, volumeName string) (err error) {
```

Comment:
`handleAttachmentTicketCreation/Deletion` is quite similar in related controllers. Can we make them shareable to reduce duplicated codes?

Reply:
We discussed this with Derek and Shuo. This is on my todo list for part-2 PR together with upgrade path